ModConfigMenu/ModConfigMenu/Src/ModConfigMenuAPI/Classes/MCM_API_SettingsPage.uc

```diff
@@ -6,17 +6,36 @@ function int GetPageId();
 
 delegate SaveStateHandler(MCM_API_SettingsPage SettingsPage);
 
+delegate VoidSettingHandler(MCM_API_Setting Setting, name SettingName);
+delegate BoolSettingHandler(MCM_API_Setting Setting, name SettingName, bool SettingValue);
+delegate FloatSettingHandler(MCM_API_Setting Setting, name SettingName, float SettingValue);
+delegate StringSettingHandler(MCM_API_Setting Setting, name SettingName, string SettingValue);
+
 function SetPageTitle(string title);
 
 // By default Save/Cancel buttons are not visible, you can choose to use them.
-function EnableSaveAndCancelButtons(delegate<SaveStateHandler> SaveHandler, delegate<SaveStateHandler> CancelHandler);
+function SetSaveHandler(delegate<SaveStateHandler> SaveHandler);
+function SetCancelHandler(delegate<SaveStateHandler> CancelHandler);
+
 // By default Reset button is not visible, you can choose to use it.
-function EnableResetToDefaultButton(delegate<SaveStateHandler> Handler);
+function EnableResetToDefaultButton(delegate<SaveStateHandler> ResetHandler);
 
 // It's done this way because of some UI issues where dropdowns don't behave unless you initialize
 // the settings UI widgets from bottom up.
-function array<MCM_API_Setting> MakeSettings(array<name> SettingNames);
+function MCM_API_Label AddLabel(name SettingName, string Label, string Tooltip);
+function MCM_API_Button AddButton(name SettingName, string Label, string Tooltip, string ButtonLabel, delegate<VoidSettingHandler> Handler);
+function MCM_API_Checkbox AddCheckbox(name SettingName, string Label, string Tooltip, bool InitiallyChecked, delegate<BoolSettingHandler> Handler);
+function MCM_API_Slider AddSlider(name SettingName, string Label, string Tooltip, float SliderMin, float SliderMax, float SliderStep, float InitialValue, delegate<FloatSettingHandler> Handler);
+function MCM_API_Spinner AddSpinner(name SettingName, string Label, string Tooltip, array<string> Options, string Selection, delegate<StringSettingHandler> Handler);
+function MCM_API_Dropdown AddDropdown(name SettingName, string Label, string Tooltip, array<string> Options, string Selection, delegate<StringSettingHandler> Handler);

```

Comment:
These should exist in a separate helper class _(something like `MCM_API_ControlFactory`)_, they don't belong here.  Instead there should be a single `AddControl(MCM_API_Setting)` method.

This allows us (or users) to add new controls in the future, without having to change the API for this class.